tests/integration/telemetry/tracing/stackdriver/tracing_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2019 Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"testing"
+	"time"
+
+	"fmt"
+
+	"os"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/bookinfo"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+	util "istio.io/istio/tests/integration/mixer"
+)
+
+var (
+	ist            istio.Instance
+	bookinfoNsInst namespace.Instance
+	galInst        galley.Instance
+	ingInst        ingress.Instance
+)
+
+// TestProxyTracing exercises the trace generation features of Istio, based on the Envoy Trace driver for stackdriver.
+// The test verifies that all expected spans (a client span and a server span for each service call in the sample bookinfo app)
+// are generated and that they are all a part of the same distributed trace with correct hierarchy and name.
+//
+// More information on distributed tracing can be found here: https://istio.io/docs/tasks/telemetry/distributed-tracing/stackdriver/
+func TestProxyTracing(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			t.Logf("os STACKDRIVER_TRACING_MAX_NUMBER_OF_ATTRIBUTES %v", os.Getenv("STACKDRIVER_TRACING_MAX_NUMBER_OF_ATTRIBUTES"))
+			retry.UntilSuccessOrFail(t, func() error {
+				util.SendTraffic(ingInst, t, "Sending traffic", "", "", 1)
+
+				return fmt.Errorf("cannot get traces from stackdriver %v", bookinfoNsInst)
```

Comment:
how is this testing stack driver?

Reply:
It had tracing enabled through stackdriver but it's not complete yet and I will add the test in a different PR.
Those files were added by a bad commit.. they are removed now